tasks/main.yml

```diff
@@ -67,12 +68,11 @@
       # Ansible's markers to indicate it's a machine-managed file.
       blockinfile:
         path: "{{ nix_config_dir.path }}/nix.conf"
-        block: "experimental-features = 'nix-command flakes'"
+        block: "experimental-features = nix-command flakes"
         insertafter: EOF
         owner: "{{ nix_user }}"
         mode: "0644"
         create: true
         state: present
   when:  # noqa no-handler
-    - nix_directory is changed
     - nix_flakes
```

Comment:
Actually, please ignore this comment. I think that more work must be done in this task. I'm working on a solution now.

Reply:
If we can resolve the trouble I'm having with the `when: nix_directory is changed` condition on a Nix-naive Ubuntu VM, I think your suggestion is good! But I'd like to hear your own results first before making this commit.

Reply:
Ok, so as mentioned above, you are right, the `nix_directory is changed` bit needs to go. What we _should_ do is instead see if we've already written the `nix-command flakes` bit in the config file (or if the config file even exists). Ideally, it would be nice if we could run a nix command to tell us if flakes have been configured, but I'm not aware of one (and I'm not very knowledgeable about nix, so please let me know if I'm missing something here). I tried `nix-env --query --installed`, but it doesn't list flakes there, so I guess it isn't treated like other packages.

Anyways, I've implemented these checks in https://github.com/Ableton/ansible-role-nix/pull/74/commits/7d5d60786fb574855d2b450793d4a293644a6e36. With this change, we should only need the `when: nix_flakes` condition on that `block`.